CONTRIBUTING.md

```diff
@@ -422,25 +422,35 @@ Note: If you have run `test/ipynb/mpl_tester.ipynb` locally it is possible some
 
 ## Style and lint
 
-Qiskit Terra uses 2 tools for verify code formatting and lint checking. The
+Qiskit Terra uses three tools for verify code formatting and lint checking. The
 first tool is [black](https://github.com/psf/black) which is a code formatting
 tool that will automatically update the code formatting to a consistent style.
 The second tool is [pylint](https://www.pylint.org/) which is a code linter
 which does a deeper analysis of the Python code to find both style issues and
-potential bugs and other common issues in Python.
-
-You can check that your local modifications conform to the style rules
-by running `tox -elint` which will run `black` and `pylint` to check the local
-code formatting and lint. If black returns a code formatting error you can
-run `tox -eblack` to automatically update the code formatting to conform to
-the style. However, if `pylint` returns any error you will have to fix these
-issues by manually updating your code.
-
-Because `pylint` analysis can be slow, there is also a `tox -elint-incr` target, which only applies
-`pylint` to files which have changed from the source github. On rare occasions this will miss some
-issues that would have been caught by checking the complete source tree, but makes up for this by
-being much faster (and those rare oversights will still be caught by the CI after you open a pull
-request).
+potential bugs and other common issues in Python. The third tool is the linter
+[ruff](https://github.com/charliermarsh/ruff), which has been recently
+introduced into Qiskit Terra on an experimental basis. Only a very small number
+of rules are enabled.
+
+You can check that your local modifications conform to the style rules by
+running `tox -elint` which will run `black`, `ruff`, and `pylint` to check the
+local code formatting and lint. If black returns a code formatting error you can
+run `tox -eblack` to automatically update the code formatting to conform to the
+style. However, if `ruff` or `pylint` return any error you will have to fix
```

Comment:
Ruff does have some auto fix rules which can automatically be fixed (to some degree at least) with `--fix`. But, probably not worth getting into that here.

Reply:
No need to block this PR on this. But, I find it useful to have three top-level commands:

* `tox -e fmt`, run all formatters
* `tox -e fix`, run all formatters & fixers
* `tox -e lint`, run all formatters and fixers in check-only mode, along with pure linters

Why split `fix` and `fmt`? You sometimes only want to format your code without making any semantic changes. For example, some "fixers" will remove unused import. But that import might only be unused because you are still writing the code.

We've used `fmt` vs `fix` vs `lint` in Pantsbuild (a build system) for the past year and it's worked great.

Reply:
And we have unused imports just for the side effects. I made a comment about fixing below. I agree you have to be a bit more careful with running fix. I think something like your three commands looks good. But maybe not for this PR.